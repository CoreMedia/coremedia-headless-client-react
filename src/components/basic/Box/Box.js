// @flow
import styled from 'styled-components';

const Box = styled.div`
  margin: 0;
  padding: 0;
  font-family: ${props => props.theme.typography.fontFamily.text};
  line-height: 1.4;
  box-sizing: border-box;
`;

Box.displayName = 'Box';

Box.defaultProps = {
  theme: {
    typography: {
      fontFamily: {
        text: 'Helvetica',
      },
    },
  },
};

export default Box;
