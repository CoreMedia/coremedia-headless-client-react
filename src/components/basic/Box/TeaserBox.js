// @flow
import styled from 'styled-components';

const TeaserBox = styled.div`
  position: relative;
  border: 0;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  > * {
    box-sizing: border-box;
  }
`;
TeaserBox.displayName = 'TeaserBox';

export default TeaserBox;
