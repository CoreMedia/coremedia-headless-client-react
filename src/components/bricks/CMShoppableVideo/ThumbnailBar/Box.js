// @flow
import styled from 'styled-components';

const ThumbnailsBox = styled.div`
  display: inline-block;
  white-space: nowrap;
`;

ThumbnailsBox.displayName = 'ThumbnailsBox';

export default ThumbnailsBox;
