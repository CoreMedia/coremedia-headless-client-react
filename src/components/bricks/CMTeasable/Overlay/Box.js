// @flow
import styled from 'styled-components';
import { media, getColors } from '../../../../styles/themes/utils';

const Box = styled.div`
  color: ${props => getColors(props.theme.colors, props.color).fg};
  background-color: ${props => getColors(props.theme.colors, props.color).bgsolid};
  position: relative;
  overflow: hidden;
  padding: 1em;
  max-width: 100%;
  max-height: 100%;
  ${props => media(props.theme.breakpoints.tablet)`
    background-color: ${props => getColors(props.theme.colors, props.color).bgtrans};
    position: absolute;
    width: ${props => props.width}%;
    height: ${props => props.height};
    left: ${props => (100 - props.width) / 2}%;
    bottom: ${props => props.bottom};
  `};
`;

Box.displayName = 'Box';

Box.defaultProps = {
  theme: {
    colors: {
      red: {
        fg: '#fafafa',
        bgsolid: 'rgba(221, 52, 40, 1)',
        bgtrans: 'rgba(221, 52, 40, 0.7)',
        huerotation: '320deg',
      },
      default: {
        fg: '#fafafa',
        bgsolid: 'rgba(239, 223, 15, 1)',
        bgtrans: 'rgba(239, 223, 15, 0.7)',
        huerotation: '5deg',
      },
    },
    breakpoints: {
      tablet: 768,
    },
  },
};

export default Box;
