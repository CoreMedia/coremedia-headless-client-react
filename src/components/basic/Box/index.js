// @flow
import Box from './Box';
import TeaserBox from './TeaserBox';

export { Box, TeaserBox };
