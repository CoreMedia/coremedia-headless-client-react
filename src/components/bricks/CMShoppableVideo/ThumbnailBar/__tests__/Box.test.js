// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Box from '../Box';

describe('Box Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Box>Content</Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
