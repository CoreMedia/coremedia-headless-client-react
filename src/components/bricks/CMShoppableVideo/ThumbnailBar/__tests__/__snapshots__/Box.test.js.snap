// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component should render correctly 1`] = `
.c0 {
  display: inline-block;
  white-space: nowrap;
}

<div
  className="c0"
>
  Content
</div>
`;
